### PR TITLE
Support code block part in inline chat

### PR DESCRIPTION
**Pull request by aeschli**
- extract codeBlockPart

### CODE REVIEW

File: src/vs/workbench/contrib/chat/browser/codeBlockPart.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+import { getSimpleEditorOptions } from 'vs/workbench/contrib/codeEditor/browser/simpleEditorOptions';
+
+const $ = dom.$;
+
+
+export interface ICodeBlockData {
+	text: string;
+	languageId: string;
+	codeBlockIndex: number;
+	element: unknown;
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/codeBlockPart.ts:46]: I don't like this, can you make it optional but typed, or a generic parameter, or something?

Comment (roblourens) [reply src/vs/workbench/contrib/chat/browser/codeBlockPart.ts:46]: Eh, I see what you do with `IChatCodeBlockActionContext`, would be nice to have more type support here but it's ok if it doesn't work well

**Review by roblourens — APPROVED** [review]

**Comment by roblourens** [comment]
Please cc me on chat renderer PRs

File: src/vs/workbench/contrib/chat/browser/codeBlockPart.ts (hunk 1 above)
Comment (aeschli) [reply src/vs/workbench/contrib/chat/browser/codeBlockPart.ts:46]: @roblourens I agree that this isn't nice. I'd love to make this better. I ping you today and we can brainstorm.

**Pull request merged**